lib/galaxy/datatypes/media.py

```diff
@@ -183,7 +269,10 @@ class Mkv(Video):
     def sniff(self, filename: str) -> bool:
         if which("ffprobe"):
             metadata, streams = ffprobe(filename)
-            return "matroska" in metadata["format_name"].split(",")
+            vp_check = any(
+                stream["codec_name"] in ["vp8", "vp9"] for stream in streams if stream["codec_type"] == "video"
+            )
+            return _get_file_format_from_magic_number(filename, "mkv") and not vp_check
```

Comment:
I don't think this restriction makes sense for mkvs, they should be able to host vp8/9 encoded streams.

But this is also somewhat a general 'problem' with the datatypes in that we describe both containers and encodings, and we don't capture that metadata in galaxy... but on the other hand I'm not sure I've encountered many tools that are doing video processing and aren't using something like `ffmpeg` / `libav` / etc and would be able to handle different stream codecs within a container.

Do you have a reason to add this restriction here? If so, perhaps that would be best expressed as metadata of the file, and then annotate that metadata here, and whatever tools you use can make decisions based on that metadata. 

Also I'm not sure of a benefit we get from this specific change; we're still doing the `ffprobe` so we aren't doing any less computation via using magic numbers (i.e. the usual reason for adding them), are magic numbers more accurate than checking for `matroska`?

Reply:
The problem is that for `mkv` and `webm` formats, they look the same to `ffprobe` and have the same magic number. So the way that I found is that `webm` video codec could only be vp8 or vp9, That is why I am using this approach. And for computation you are right it doesn't make any difference that is why I put the mkv and webm at the end of the sniffer tags in `datatypes_conf.xml.sample`.

Reply:
Yes, you are correct, webms *are* mkvs, just with restricted format lists. I guess there is precedent for datatypes that are subsets (bed)

But in this case we would potentially (wrongly?) detect a user's uploaded `mkv` that happens to use vp8 as `webm` format, which is an odd scenario to encounter.

I'm not sure what the best behaviour is here and I'd be happy to hear input from other folks who deal with video data.

Reply:
There is also another way to find out the format in Python and it is with magic but I'm afraid it will slow the process.
```
>>> import magic
>>> f = magic.Magic(mime=True)
>>> f.from_file('testfile')
```